subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskContainer.java

```diff
@@ -611,18 +611,18 @@ public void remove() {
 
     @Override
     public Action<? super Task> whenObjectRemoved(Action<? super Task> action) {
-        warnAboutRemoveMethodDeprecation("whenObjectRemoved(Action)");
+        DeprecationLogger.nagUserOfDiscontinuedMethod("TaskContainer.whenObjectRemoved(Action)");
         return super.whenObjectRemoved(action);
     }
 
     @Override
     public void whenObjectRemoved(Closure action) {
-        warnAboutRemoveMethodDeprecation("whenObjectRemoved(Closure)");
+        DeprecationLogger.nagUserOfDiscontinuedMethod("TaskContainer.whenObjectRemoved(Closure)");
         super.whenObjectRemoved(action);
     }
 
     private void warnAboutRemoveMethodDeprecation(String methodName) {
-        DeprecationLogger.nagUserOfDiscontinuedMethod("TaskContainer." + methodName, "Prefer disabling the task instead, see Task.setEnabled(boolean).");
+        DeprecationLogger.nagUserOfDeprecated("Using TaskContainer." + methodName + " to remove tasks", "Prefer disabling the task instead, see Task.setEnabled(boolean).");
```

Comment:
Shouldn't we try to use `nagUserOfDiscontinuedMethodInvocation()` here?  (note this is a different method from `nagUserOfDiscontinuedMethod` and does pretty much exactly what we're doing in this method)  May need to add an override to provide advice.

Reply:
Thanks, that's a better fit. there are a billion choices in `DeprecationLogger` nowadays.